src/vs/platform/telemetry/common/telemetryService.ts

```diff
@@ -108,6 +108,11 @@ export class TelemetryService implements ITelemetryService {
 			return Promise.resolve(undefined);
 		}
 
+		// Check if event name contains a number and throw as the GDPR catalog doesn't play nice with numbers in event names
+		if (/\d/.test(eventName)) {
+			throw new Error(`Event name '${eventName}' contains a number. Event names should not contain numbers.`);
```

Comment:
what's the purpose of this error? to let us know we've made a mistake? can we do this with tooling instead so we aren't polluting the error console for no reason?

Reply:
I can do this in the extractor instead if we prefer, it's just to let you know the telemetry event will not be sent through even if it properly is logged in the output channel etc. Maybe it's not super helpful though because telemetry is disabled when developing 🤔

Reply:
yea either a linting rule or the extractor seems sufficient to me